src/transformers/pytorch_utils.py

```diff
@@ -44,6 +39,14 @@
 is_torch_greater_or_equal_than_1_12 = parsed_torch_version_base >= version.parse("1.12")
 
 
+if is_torch_greater_or_equal_than_2_4:
+    from torch.distributed.tensor import Replicate
+    from torch.distributed.tensor.parallel import (
+        ColwiseParallel,
+        RowwiseParallel,
+    )
```

Comment:
Actually, `Replicate` is in `torch.distributed.tensor` >= torch 2.5. 
Between 2.0-2.4 (included), it is in `torch.distributed._tensor`.

Thus it seems there are two options:
Option 1:
```
try:
    from torch.distributed.tensor import Replicate
except ImportError:
    from torch.distributed._tensor import Replicate
```
Option 2:
bump the requirement to 2.5.

`ColwiseParallel` and `RowwiseParallel` well exists since 2.0.

Reply:
Hi @kwen2501 - no preference from my side.  I have a PR that does the first option here: https://github.com/huggingface/transformers/pull/34816

But happy to abandon or switch in favor of any you have to add support.